mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -271,6 +271,13 @@ public long getHighestModSeq(Mailbox mailbox) throws MailboxException {
     public MessageMetaData add(Mailbox mailbox, MailboxMessage message) throws MailboxException {
         CassandraId mailboxId = (CassandraId) mailbox.getMailboxId();
 
+        save(mailbox, enrichMessage(message, mailboxId))
+            .thenCompose(voidValue -> indexTableHandler.updateIndexOnAdd(message, mailboxId))
+            .join();
+        return new SimpleMessageMetaData(message);
+    }
+
+    private MailboxMessage enrichMessage(MailboxMessage message, CassandraId mailboxId) throws MailboxException {
```

Comment:
do we really modify the input object ?
And the method name is not precise enough : enrich doesn't tell me what information I will get from this call (it should)